libp2p/src/main/kotlin/io/libp2p/mux/yamux/YamuxHandler.kt

```diff
@@ -34,19 +34,25 @@ open class YamuxHandler(
         private val buffered = ArrayDeque<ByteBuf>()
 
         fun add(data: ByteBuf) {
-            buffered.add(data)
+            buffered.add(data.retain())
```

Comment:
Please see the update on this `retain()`: https://github.com/Peergos/jvm-libp2p/pull/12